tests/Collections/Iterators/DictionaryIteratorTest.php

```diff
@@ -173,10 +174,10 @@ public function getValueReturnedKeyValuePairTestData(): array
      */
     public function testGetValueKeyType( Dictionary $dictionary )
     {
+        $typeLookup      = TypeLookupSingleton::getInstance();
+        $iteratorKeyType = $typeLookup->getByValue( $dictionary->getIterator()->getValue()->getKey() );
         $this->assertTrue(
-            $dictionary->getKeyType()->equals(
-                $dictionary->getIterator()->getValue()->getKey()
-            ),
+            $dictionary->getKeyType()->equals($iteratorKeyType),
```

Comment:
This test seems a little weird, because there is no corresponding test for the value types. Might want to make an issue for this?

Reply:
See code comment